From d6655cca7db1b960456811b8206ce222700e010d Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Tue, 30 Jan 2018 13:19:58 -0800
Subject: [PATCH] Remove compressed responses from web servers.

While zlib is mandatory for transmission, it is not mandatory for curl.

A libcurl that has been compiled with no support for zlib will return no data if compressed responses are set to on.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 libtransmission/web.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libtransmission/web.c b/libtransmission/web.c
index c7f0627..db34976 100644
--- a/libtransmission/web.c
+++ b/libtransmission/web.c
@@ -180,7 +180,7 @@ createEasy (tr_session * s, struct tr_web * web, struct tr_web_task * task)
   task->timeout_secs = getTimeoutFromURL (task);
 
   curl_easy_setopt (e, CURLOPT_AUTOREFERER, 1L);
-  curl_easy_setopt (e, CURLOPT_ENCODING, "gzip;q=1.0, deflate, identity");
+  curl_easy_setopt (e, CURLOPT_ENCODING, "");
   curl_easy_setopt (e, CURLOPT_FOLLOWLOCATION, 1L);
   curl_easy_setopt (e, CURLOPT_MAXREDIRS, -1L);
   curl_easy_setopt (e, CURLOPT_NOSIGNAL, 1L);
-- 
2.7.4

